tests/cli-tests/compression/basic.sh

```diff
@@ -24,6 +24,12 @@ zstd -c file       | zstd -t
 zstd --stdout file | zstd -t
 println bob | zstd | zstd -t
 
+# Test keeping input file when compressing to stdout in gzip mode
+ln -s "$(type -p zstd)" ./gzip
+./gzip -c file       | zstd -t ; test -f file
```

Comment:
minor :
if you want to test that both sides of the test are completing fine, prefer `&&` instead of `;` separator.

```
> false ; echo this is successful
this is successful
> false ; true
> echo $?
0

> false && echo this is successful

> false && true
> echo $?
1
```

Reply:
Please test with `set -e` as these scripts are using. In my experience the behavior of that with `&&` is odd - `&&` similar to `if` and `while` will temporarily turn off errexit mode and simply hide the error. try this example script:

````
$ cat x.sh 
set -e

false | false && true && echo it detected the failure
true && echo "did it really find the false?"
$ sh x.sh ; echo $?
did it really find the false?
0
```